compiler/lib/src/main/scala/ast/Ast.scala

```diff
@@ -587,7 +611,63 @@ object Ast {
 
   }
 
-  /** Port matching specifier */
+  /** Initial state Specifier */
+  final case class SpecInitial(
+    action: Option[Ident],
+    state: Ident
+  )
+
+  /** State definition */
+  final case class DefState(
+    name: Ident,
+    members: Option[List[StateMember]]
+  )
+
+  /** Action definition */
+  final case class DefAction(
+    name: Ident,
+    typeName: Option[QualIdent]
+  )
+
+  /** Guard definition */
+  final case class DefGuard(
+    name: Ident,
+    typeName: Option[QualIdent]
+  )
+
+  /** Junction definition */
+  final case class DefJunction(
```

Comment:
This doesn't follow the syntax in the spec, which is described in terms of enter expressions. Let's follow the spec.